jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -676,10 +850,30 @@ def _import_spectral_regions(self, spec_region, mode=NewMode, viewer=None):
                 if index == 0:
                     m = NewMode
                 else:
-                    m = mode[index - 1]
+                    m = SUBSET_MODES[mode[index - 1]]
+            elif mode:
+                m = SUBSET_MODES[mode]
             else:
-                m = mode
+                m = SUBSET_MODES[self.combination_mode.selected]
             self.app.session.edit_subset_mode.mode = m
             s = RangeSubsetState(lo=sub_region.lower.value, hi=sub_region.upper.value,
-                                 att=range_viewer.slice_component_label)
+                                 att=range_viewer.state.x_att)
             range_viewer.apply_subset_state(s)
+
+    def _apply_subset_state_to_viewer(self, state, viewer):
+        self.app.session.edit_subset_mode.mode = SUBSET_MODES[self.combination_mode.selected]
+        if isinstance(viewer, SpecvizProfileView):
+            raise NotImplementedError("This method is currently only for image viewers")
+        else:
+            viewer.apply_roi(state)
+        self.app.session.edit_subset_mode.edit_subset = None  # No overwrite next iteration # noqa
+
+    @observe('combination_selected')
+    def _combination_selected_updated(self, change):
+        self.app.session.edit_subset_mode.mode = SUBSET_MODES[self.combination_mode.selected]
+
+    def _update_combination_mode(self):
```

Comment:
this is only ever called when linking is changed, but needs to be two-way synced, so that changing the mode in the UI (the glue dropdown) or via internal glue API, is reflected in `self.combination_mode`

Reply:
has this been addressed?  When changing the combination mode from the UI, I do not see the internal traitlet being updated to match (although changing the `combination_mode.selected` does reflect in the UI via `_combination_selected_updated`).